modules/xmpp/JingleSessionPC.js

```diff
@@ -1662,6 +1662,40 @@ export default class JingleSessionPC extends JingleSession {
         this._addOrRemoveRemoteStream(false /* remove */, elem);
     }
 
+    /**
+     * Handles the removal of SSRCs from remote description when an endpoint leaves the call.
+     *
+     * @param {*} id Endpoint id of the participant that has left the call.
+     */
+    removeRemoteStreamsOnLeave(id) {
+        const removeSsrcInfo = this.peerconnection.getRemoteSourceInfoByParticipant(id);
```

Comment:
This is a little bit problematic, but because of the async queue the sources may not be there when this method is called, but they could be on the queue. I would move this check inside the `workFunction`

Reply:
sounds good, will move it.

Reply:
Also, Jicofo will stop sending the `source-remove` on participant leaving the call so `removeRemoteStream` task will not be queued.